### PR TITLE
Add explaination about tidb_allow_batch_cop

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### What is changed, added or deleted? (Required)

Add explaination about a new system variable `tidb_allow_batch_cop` to tidb-specific-variables.md and tiflash/tune-tiflash-performance.md

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
  - https://github.com/pingcap/tidb/pull/16030

### CODE REVIEW

**Comment by TomShawn** [comment]
@JaySon-Huang 这个 PR 是否为 docs-special-week 的 PR 呢？如果是，请将 branch rebase 到 docs-special-week。

**Comment by JaySon-Huang** [comment]
@TomShawn 有一个新功能开发完，在4.0GA会带出去，但是之前没有补充相关的文档说明。没有在 special-week 原计划内。

File: tidb-specific-system-variables.md [hunk 1]
```diff
@@ -736,3 +736,13 @@ TiDB 默认会在建表时为新表分裂 Region。开启该变量后，会在
...
 
 这个变量用来显示上一个 `execute` 语句所使用的执行计划是不是直接从 plan cache 中取出来的。
+
+### tidb_allow_batch_cop <span class="version-mark">从 v4.0 版本开始引入</span>
+
+作用域：SESSION
+
+默认值：0
+
+这个变量用来设置从 TiFlash 读取时，是否把 region 的请求进行合并。
```
Comment (TomShawn) [inline tidb-specific-system-variables.md:746]:
```suggestion
这个变量用来设置从 TiFlash 读取数据时，是否把 Region 的请求进行合并。
```

File: tidb-specific-system-variables.md [hunk 2]
```diff
@@ -736,3 +736,13 @@ TiDB 默认会在建表时为新表分裂 Region。开启该变量后，会在
...
+
+当查询的表拥有 TiFlash 副本，且查询涉及的 region 数量比较大，可以尝试设置该变量为 1：对带 aggregation 的 TiFlash coprocessor 请求生效；或设置该变量为 2：对全部的 TiFlash coprocessor 请求生效。
```
Comment (TomShawn) [inline tidb-specific-system-variables.md:748]:
```suggestion
当查询的表拥有 TiFlash 副本，且查询涉及的 Region 数量比较多，可以尝试设置该变量为 `1`（对带 `aggregation` 的 TiFlash coprocessor 请求生效），或设置该变量为 `2`（对全部的 TiFlash coprocessor 请求生效）。
```

File: tiflash/tune-tiflash-performance.md [hunk 3]
```diff
@@ -22,10 +22,30 @@ aliases: ['/docs-cn/dev/reference/tiflash/tune-performance/']
     set @@tidb_distsql_scan_concurrency = 80;
     ```
 
-2. 开启聚合推过 JOIN / UNION 等 TiDB 算子的优化：
+2. 开启 Super batch 功能：  
+这个变量用来设置从 TiFlash 读取时，是否把 region 的请求进行合并。当查询中涉及的 region 数量比较大，可以尝试设置该变量为 1：对带 aggregation 的 TiFlash coprocess 请求生效；设置该变量为 2：对全部的 TiFlash coprocessor 请求生效。
```
Comment (TomShawn) [inline tiflash/tune-tiflash-performance.md:26]: + `这个变量`是什么变量呢？请明确写出变量名，最好能附上 `tidb-specific-system-variables.md` 里的链接。
+ region 的首字母为大写。

File: tiflash/tune-tiflash-performance.md [hunk 4]
```diff
@@ -22,10 +22,30 @@ aliases: ['/docs-cn/dev/reference/tiflash/tune-performance/']
...
+这个变量用来设置优化器是否执行聚合函数下推到 Join 之前的优化操作。当查询中聚合操作执行很慢时，可以尝试设置该变量为 1。
 
     {{< copyable "sql" >}}
 
     ```sql
     set @@tidb_opt_agg_push_down = 1;
     ```
+
+4. 尝试开启 `Distince` 推过 `Join` / `Union` 等 TiDB 算子的优化：
+这个变量用来设置优化器是否执行带有 `Distinct` 的聚合函数（比如 `select count(distinct a) from t`）下推到 Coprocessor 的优化操作。
```
Comment (TomShawn) [inline tiflash/tune-tiflash-performance.md:44]: `这个变量`是什么变量呢？请明确写出变量名，最好能附上 `tidb-specific-system-variables.md` 里的链接。

File: tiflash/tune-tiflash-performance.md [hunk 5]
```diff
@@ -22,10 +22,30 @@ aliases: ['/docs-cn/dev/reference/tiflash/tune-performance/']
...
+当查询中带有 `Distinct` 的聚合操作执行很慢时，可以尝试设置该变量为 `1`。
```
Comment (TomShawn) [inline tiflash/tune-tiflash-performance.md:45]: 如需换行，请在两行之间插入空行，否则会显示无换行。

File: tidb-specific-system-variables.md [hunk 6]
```diff
@@ -736,3 +736,13 @@ TiDB 默认会在建表时为新表分裂 Region。开启该变量后，会在
 默认值：0
 
 这个变量用来显示上一个 `execute` 语句所使用的执行计划是不是直接从 plan cache 中取出来的。
+
+### tidb_allow_batch_cop <span class="version-mark">从 v4.0 版本开始引入</span>
+
+作用域：SESSION
```
Comment (hanfei1991) [inline tidb-specific-system-variables.md:742]: SESSION + GLOBAL 都支持的

Comment (JaySon-Huang) [inline tidb-specific-system-variables.md:742]:
```suggestion
作用域：SESSION | GLOBAL
```

Comment (JaySon-Huang) [reply tidb-specific-system-variables.md]: Addressed

**Comment by hanfei1991** [comment]
LGTM

File: tidb-specific-system-variables.md [hunk 7]
```diff
@@ -736,3 +736,13 @@ TiDB 默认会在建表时为新表分裂 Region。开启该变量后，会在
 默认值：0
 
 这个变量用来显示上一个 `execute` 语句所使用的执行计划是不是直接从 plan cache 中取出来的。
+
+### tidb_allow_batch_cop <span class="version-mark">从 v4.0 版本开始引入</span>
```
Comment (yikeke) [inline tidb-specific-system-variables.md:740]: This change will be overwritten by docs-special-week changes. Please check the latest version of this file.

Comment (yikeke) [reply tidb-specific-system-variables.md:740]: The latest doc is https://pingcap.com/docs-cn/dev/tidb-specific-system-variables/#tidb_allow_batch_cop-%E4%BB%8E-v40-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5

Comment (yikeke) [reply tidb-specific-system-variables.md:740]: Confirmed with @JaySon-Huang : will keep the change of https://github.com/pingcap/docs-cn/pull/3159/files#diff-36f1f55efb0294795b77879ecad344a0R869

**Pull request merged**